html2docx/html2docx.py

```diff
@@ -70,6 +70,7 @@ def __init__(self, title: str):
         self.doc.core_properties.title = title
         self.list_style: List[str] = []
         self.href = ""
+        self.tag: Optional[str] = None
```

Comment:
I don’t think remembering `self.tag` is needed: initialize `self.table_data = None`. When encountering a table, set `self.table_data = []`. At the end of a table, set `self.table_data = None`.
`th` and `td` can then assume `self.table_data` is a list and fill it, `tr` can also add a new list to `self.table_data`.